### PR TITLE
docs(wireguard-broker): add docs and examples

### PR DESCRIPTION
**Pull request by DavidNiehues**
This PR, majority of the contributions by @aminfa, adds tests, documentation and examples to the wireguard-broker crate.

### CODE REVIEW

File: wireguard-broker/src/api/msgs.rs [hunk 1]
```diff
@@ -17,25 +25,43 @@ pub struct Envelope<M: AsBytes + FromBytes> {
...
 
+    /// Gets the interface specification as a `&str`.
+    ///
+    /// # Errors
+    /// Returns a [Utf8Error] if the interface specification isn't utf8 encoded.
     pub fn iface(&self) -> Result<&str, Utf8Error> {
         from_utf8(self.iface_bin())
     }
 
+    /// Sets the interface specification to `iface`. No check is made wether `icace` is correctly
```
Comment (aparcar) [inline wireguard-broker/src/api/msgs.rs:59]: Typo icace

File: wireguard-broker/src/api/msgs.rs [hunk 2]
```diff
@@ -47,17 +73,24 @@ impl SetPskRequest {
...
+    ///
+    /// # Result
+    /// Returns [None] if `iface` is longer than 255 bytes. Otherwise, it returns
+    /// [Some(())](Some).
     pub fn set_iface(&mut self, iface: &str) -> Option<()> {
         self.set_iface_bin(iface.as_bytes())
     }
 }
 
+/// Message format for response top the set pre-shared key operation.
```
Comment (aparcar) [inline wireguard-broker/src/api/msgs.rs:86]: Typo top

File: wireguard-broker/src/api/msgs.rs (hunk 1 above)
Comment (phildremi) [reply wireguard-broker/src/api/msgs.rs:59]: Also: `whether`

Comment (aparcar) [reply wireguard-broker/src/api/msgs.rs:59]: Seemingly wether is an English word with the same meaning as whether :)

File: wireguard-broker/src/api/server.rs [hunk 3]
```diff
@@ -53,22 +78,28 @@ where
...
         let msgs::MsgType::SetPsk = typ; // Assert type
 
-        let req = zerocopy::Ref::<&[u8], Envelope<SetPskRequest>>::new(req)
-            .ok_or(BrokerServerError::InvalidMessage)?;
-        let mut res = zerocopy::Ref::<&mut [u8], Envelope<SetPskResponse>>::new(res)
-            .ok_or(BrokerServerError::InvalidMessage)?;
+        let req =
+            zerocopy::Ref::<&[u8], Envelope<SetPskRequest>>::new(req).ok_or(InvalidMessage)?;
+        let mut res =
+            zerocopy::Ref::<&mut [u8], Envelope<SetPskResponse>>::new(res).ok_or(InvalidMessage)?;
```
Comment (aparcar) [inline wireguard-broker/src/api/server.rs:84]: How come the formatting here changes? Didn't we use `cargo fmt` before, too?

File: wireguard-broker/src/api/server.rs [hunk 4]
```diff
@@ -95,3 +126,60 @@ where
         Ok(())
     }
 }
+
+// We can only include this test if this feature is enabled, because otherwise
+// brokers::netlink::SetPskError is not defined.
+#[cfg(all(feature = "experiment_api", target_os = "linux"))]
```
Comment (aparcar) [inline wireguard-broker/src/api/server.rs:132]: Since we mock the WireGuard broker anyway, shouldn't this be able to run on non-linux, too?

**Comment by aparcar** [comment]
Thanks again (and again) for your work, I just found some minor things and I wonder about the re-formating of the zerocopy lines.

File: wireguard-broker/src/api/server.rs (hunk 4 above)
Comment (DavidNiehues) [reply wireguard-broker/src/api/server.rs:132]: Indeed, will fix this in the next commit/amendment to the last commit.

File: wireguard-broker/src/api/server.rs (hunk 3 above)
Comment (aparcar) [reply wireguard-broker/src/api/server.rs:84]: Nevermind this detail.

**Comment by aparcar** [comment]
All right, jokes on me. The test actually imports netlink which fails on macOS. Please revert it to the prior state and squash the two commits with the same subject. I'd do it but you did not allow maintainers to push into your branch.

**Comment by DavidNiehues** [comment]
> All right, jokes on me. The test actually imports netlink which fails on macOS. Please revert it to the prior state and squash the two commits with the same subject. I'd do it but you did not allow maintainers to push into your branch.

@aparcar  Fixed now

**Review by aparcar — APPROVED** [review]

**Pull request merged**